contracts/Vault2.sol

```diff
@@ -1,6 +1,26 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 
-contract Vault2 {
-    
+contract Vault2 is ERC20 {
+  address private owner;
+
+  mapping (address => uint) public users;
+
+  constructor() ERC20("Vault", "VAULT") {
+    owner = msg.sender;
+    _mint(msg.sender, 10000 * 10 ** uint(decimals()));
+  }
+
+  function mint() public payable {
+    uint etherSent = msg.value / 10 ** uint(decimals());
```

Comment:
Best to have these values as wei itself rather than converting it to ether, because if someone here sends 0.99 ether it will be rounded off to 0